src/pip/_internal/cli/req_command.py

```diff
@@ -325,6 +325,7 @@ def make_resolver(
             install_req_from_req_string,
             isolated=options.isolated_mode,
             use_pep517=use_pep517,
+            config_settings=getattr(options, "config_settings", None),
```

Comment:
Would it make sense to use as default value an empty dictionary?

The text in the [PEP](https://peps.python.org/pep-0517/#config-settings) mentions:
> This argument, which is passed to all hooks, is an arbitrary dictionary

```suggestion
            config_settings=getattr(options, "config_settings", {}),
```

Reply:
I considered this. But the default value [in the PEP](https://peps.python.org/pep-0517/#build-wheel) is the `config_settings` argument is not passed is set to `None`, so I decided to use `None` if the user didn't specify anything. That way, "user didn't specify anything" should look identical to the backend as the front end not passing `config_settings` at all. That way, if there are any backends out there (which I doubt there are, TBH) which don't support `config_settings` and require it to be `None` (not passed), then we won't break them.

Reply:
Yes, that makes total sense. Thank you very much for the clarification.